### PR TITLE
Rewrite initial link parsing

### PR DESCRIPTION
**Pull request by akalin-keybase**
It was failing on links with query parameters.

Rewrite it with proper URL parsing.

### CODE REVIEW

File: shared/actions/profile/index.js [hunk 1]
```diff
@@ -224,7 +224,9 @@ function outputInstructionsActionLink(): Constants.OutputInstructionsActionLink
 }
 
 function* _onAppLink(action: AppLink): SagaGenerator<any, any> {
-  const match = action.payload.link.match(/^https:\/\/keybase\.io\/(\w+)$/)
+  // TODO: When https://github.com/facebook/react-native/issues/16434
+  // is fixed, use window.URL.
+  const match = action.payload.link.match(/^https:\/\/keybase\.io\/(\w+)(\?[^?]*)(#[^#]*)?$/)
```
Comment (buoyad) [inline shared/actions/profile/index.js]: Should add `/i` flag to this so it's case insensitive, and also have (http(s)) optional.

Perhaps `/^(http(s)?:\/\/)?keybase\.io\/(\w+)(\?[^?]*)?(#[^#]*)?$/i`

Comment (buoyad) [reply shared/actions/profile/index.js]: Also in testing I noticed that navigating to `https://keybase.io/chris` works, but `https://keybase.io/Chris` leads to an empty profile. Not sure what's causing that...

Comment (akalin-keybase) [reply shared/actions/profile/index.js]: Decided to instead pull in a URL parsing library, per @chrisnojima. Also added some tests. Should handle pretty much anything we throw at it now.

**Comment by akalin-keybase** [comment]
Now with a URL parser and tests! PTAL.

**Comment by akalin-keybase** [comment]
@MarcoPolo and I spent some time trying to get iOS/Android working with new.target, but no luck. Instead, I'll try a postinstall script.

**Comment by chrisnojima** [comment]
Let me take a look at this in the morning before you merge

**Comment by akalin-keybase** [comment]
Added a postinstall hook to hack new.target out. @chrisnojima LMK if you find a better way tomorrow morning, or please review the new hook otherwise.

**Comment by chrisnojima** [comment]
Sounds good

**Comment by chrisnojima** [comment]
I have a fix for this in a PR thats off of this, ptal

**Comment by akalin-keybase** [comment]
Will merge after CI

**Pull request merged**